scipy/stats/_distn_infrastructure.py

```diff
@@ -2256,6 +2258,9 @@ def fit(self, data, *args, **kwds):
         if Narg > self.numargs:
             raise TypeError("Too many input arguments.")
 
+        if np.logical_not(np.isfinite(data)).any():
+            raise RuntimeError("The data contains np.nan values.")
```

Comment:
"contains non-finite values" is better

Reply:
changes the message in error on recent commits (in all occurrences)